pkg/catalogv2/system/system.go

```diff
@@ -285,8 +289,8 @@ func (m *Manager) install(namespace, name, minVersion, exactVersion string, valu
 		return fmt.Errorf(err.Error())
 	}
 	// Because of the behavior of `index.Get`, we need this check.
-	if isExact && chart.Version != v {
-		return fmt.Errorf("specified exact version %s doesn't exist in the index", exactVersion)
+	if v != latestVersionMatcher && chart.Version != v {
+		return fmt.Errorf("specified exact version %s doesn't exist in the index", v)
```

Comment:
> the chart.Version could be above that version since it's just a min
You are right, but the idea is that despite the naming which survives through these variables, we no longer implement actual _min version_ behaviour: both `minVersion` and `exactVersion` are treated as exact versions, with an error being thrown (from this PR on) whenever a release cannot be sourced for that version.
The current difference between `minVersion` and `exactVersion` is that `exactVersion`, when set on its own, leads to Rancher overwriting any already installed version, even if more recent, with that specific exact version.

In practice, this [does not happen](https://github.com/rancher/rancher/blob/release/v2.8/pkg/controllers/dashboard/fleetcharts/controller.go#L91) as Fleet is the only component relying on min version behaviour: only `minVersion` is specified, yet room is left for maintainers of other charts to implement different behaviour as they see fit.

TL;DR: leaving `isExact` here would mean that this condition would not apply to calls triggered by Fleet chart installs, which are the only use case for min version at the moment. Therefore, Rancher would still install a Fleet version different from that specified in `CATTLE_FLEET_VERSION` if that configured version is not found.

Reply:
Ok, I'm fine with this logic, but the comment around minVersion says: 
```
		// not enforcing exact version install, to keep any possibly newer, already installed version.
		// This prevents automated downgrades of updates done manually through the web UI, eg. for Fleet.
```
Which is either incorrect or confusing. Overall I'd like to see the above context reflected in the comments in this method.

Reply:
> Which is either incorrect or confusing.
The comment is still true, with or without this change. I think you are confused about what `chart.Version` represents. This change does not enforce that the minVersion is the version installed. It only enforces that the minVersion exists in the repository.

> Overall, I'd like to see the above context reflected in the comments in this method.
Do the changes to the function comment satisfy this request? If not, can you be more specific on what comments you are looking for.

Reply:
Would something like do?
```
@@ -281,7 +281,7 @@ func (m *Manager) install(namespace, name, minVersion, exactVersion string, valu
                isExact = true
        } else if minVersion != "" {
                v = minVersion
-               // not enforcing exact version install, to keep any possibly newer, already installed version.
+               // not setting isExact, to keep any possibly newer, already installed version.
                // This prevents automated downgrades of updates done manually through the web UI, eg. for Fleet.
        }
 
@@ -292,7 +292,10 @@ func (m *Manager) install(namespace, name, minVersion, exactVersion string, valu
                // The helm library is using github.com/pkg/errors which is deprecated
                return fmt.Errorf(err.Error())
        }
+
        // Because of the behavior of `index.Get`, we need this check.
+       // Regardless of whether `minVersion` or `exactVersion` applies, we are looking for an exact version match in
+       // the charts index.
        if v != latestVersionMatcher && chart.Version != v {
                return fmt.Errorf("specified version %s doesn't exist in the index", v)
        }
```

Reply:
To reduce confusion, I would also happily rename `isExact` to `overwriteNewer` or similar if that helps.

Reply:
That comment at top of method helps. I talked with Kevin and I mostly understand it now. I think a bigger refactor would be good, but that goes beyond scope of this issue.